soc/nordic/common/soc_secure.h

```diff
@@ -9,63 +9,63 @@
 #include <hal/nrf_ficr.h>
 
 #if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+int soc_secure_mem_read(void *dst, void *src, size_t len);
 #if NRF_GPIO_HAS_SEL
 void soc_secure_gpio_pin_mcu_select(uint32_t pin_number, nrf_gpio_pin_sel_t mcu);
 #endif
 
-int soc_secure_mem_read(void *dst, void *src, size_t len);
+#else /* defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) */
+static inline int soc_secure_mem_read(void *dst, void *src, size_t len)
+{
+	(void)memcpy(dst, src, len);
+	return 0;
+}
+#if NRF_GPIO_HAS_SEL
+static inline void soc_secure_gpio_pin_mcu_select(uint32_t pin_number, nrf_gpio_pin_sel_t mcu)
+{
+	nrf_gpio_pin_control_select(pin_number, mcu);
+}
+#endif /* NRF_GPIO_HAS_SEL */
+
+#endif /* defined CONFIG_TRUSTED_EXECUTION_NONSECURE */
 
+/* Include these soc_secure_* functions only when the FICR is mapped as secure only */
+#if defined(NRF_FICR_S)
+#if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+static inline void soc_secure_read_deviceid(uint32_t deviceid[2])
+{
+	int err;
+
+	err = soc_secure_mem_read(deviceid, (void *)&NRF_FICR_S->INFO.DEVICEID,
+				  2 * sizeof(uint32_t));
+	__ASSERT(err == 0, "Secure read error (%d)", err);
+}
 #if defined(CONFIG_SOC_HFXO_CAP_INTERNAL)
 static inline uint32_t soc_secure_read_xosc32mtrim(void)
 {
 	uint32_t xosc32mtrim;
 	int err;
 
-	err = soc_secure_mem_read(&xosc32mtrim,
-				  (void *)&NRF_FICR_S->XOSC32MTRIM,
+	err = soc_secure_mem_read(&xosc32mtrim, (void *)&NRF_FICR_S->XOSC32MTRIM,
```

Comment:
This change does not seem to be needed.

Reply:
That's is probably happened because of my code editor formatting if you find it a big issue then I can undo.

Reply:
It's not a big issue, just unnecessary noise in git blame.

Reply:
Fixed